fendermint/ipc/src/parent.rs

```diff
@@ -0,0 +1,202 @@
+    handle: Option<JoinHandle<anyhow::Result<()>>>,
+}
+
+impl PollingParentSyncer {
+    pub fn new(config: Config) -> Self {
+        Self {
+            config,
+            started: Arc::new(AtomicBool::new(false)),
+            cache: Arc::new(RwLock::new(ParentSyncerCache {
+                block_hash: RangeKeyCache::new(),
+                top_down_message: RangeKeyCache::new(),
+                membership: None,
+            })),
+            ipc_agent_proxy: Arc::new(AgentProxy {}),
+            handle: None,
+        }
+    }
+}
+
+impl ParentViewProvider for PollingParentSyncer {
+    fn latest_height(&self) -> Option<BlockHeight> {
+        self.read_cache(|cache| cache.block_hash.upper_bound())
+    }
+
+    fn block_hash(&self, height: BlockHeight) -> Option<Vec<u8>> {
+        self.read_cache(|cache| cache.block_hash.get_value(height).map(|v| v.to_vec()))
+    }
+
+    fn top_down_msgs(&self, _height: BlockHeight, nonce: Nonce) -> Vec<CrossMsg> {
+        self.read_cache(|cache| {
+            let v = cache.top_down_message.values_within_range(nonce, None);
+            // FIXME: avoid clone here, return references, let caller clone on demand
+            v.into_iter().cloned().collect()
+        })
+    }
+
+    fn membership(&self) -> Option<ValidatorSet> {
+        self.read_cache(|cache| cache.membership.clone())
+    }
+
+    fn on_finality_committed(&self, finality: &IPCParentFinality) {
+        let mut cache = self.cache.write().unwrap();
+        cache.block_hash.remove_key_till(finality.height);
+    }
+}
+
+impl Clone for PollingParentSyncer {
+    fn clone(&self) -> Self {
+        Self {
+            config: self.config.clone(),
+            started: Arc::new(AtomicBool::new(false)),
+            cache: self.cache.clone(),
+            ipc_agent_proxy: self.ipc_agent_proxy.clone(),
+            handle: None,
+        }
+    }
+}
+
+struct ParentSyncerCache {
+    block_hash: RangeKeyCache<BlockHeight, Vec<u8>>,
+    top_down_message: RangeKeyCache<Nonce, CrossMsg>,
+    membership: Option<ValidatorSet>,
+}
+
+type LockedCache = Arc<RwLock<ParentSyncerCache>>;
+
+impl PollingParentSyncer {
+    fn read_cache<F, T>(&self, f: F) -> T
+    where
+        F: Fn(&ParentSyncerCache) -> T,
+    {
+        let cache = self.cache.read().unwrap();
+        f(&cache)
+    }
+
+    /// Start the proof of finality listener in the background
+    pub fn start(&mut self) -> anyhow::Result<()> {
+        match self
+            .started
+            .compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst)
+        {
+            Ok(_) => {}
+            Err(_) => return Err(anyhow!("already started")),
+        }
+
+        let parent_syncer = self.ipc_agent_proxy.clone();
+        let config = self.config.clone();
+        let cache = self.cache.clone();
+
+        let handle =
+            tokio::spawn(async move { sync_with_parent(config, parent_syncer, cache).await });
+        self.handle = Some(handle);
```

Comment:
It looks like if the syncer fails, the `handle` will contain the error, but there will be no feedback anywhere in the logs about the error, I assume until the application is shut down and _potentially_ the `handle` is awaited upon. 

I'd say if it is allowed to fail, it should at least log the error here, but even better would be not allowing it to fail and retrying later.